### PR TITLE
Migrate to PostgreSQL

### PR DESCRIPTION
**Pull request by AAGaming00**

### CODE REVIEW

File: .github/workflows/checks.yaml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 name: Checks
 
 on:
-  pull_request_target:
-    branches:
-      - "main"
+  pull_request:
+  push:
```
Comment (gbdlin) [inline .github/workflows/checks.yaml]: Are we sure we want to move to `pull_request`? We don't have pipeline approvals enabled in this repository.

File: docker-compose.local.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ services:
     build: .
     container_name: plugin_store
     environment:
-        - DB_PATH=/app/.database/plugin_store_new.db
+        - DB_URL
```
Comment (gbdlin) [inline docker-compose.local.yml:7]: We probably want this DB_URL specified for local dev, as the postgres instance starts in the same dockerfile anyway.

File: docker-compose.yml [hunk 3]
```diff
@@ -4,26 +4,32 @@ services:
     build: .
     container_name: "${DEPLOYMENT_NAME}"
     environment:
-        - DB_PATH
+        - DB_URL
```
Comment (gbdlin) [inline docker-compose.yml:7]: Actually, same here. We can build it from `POSTGRES_PASSWORD` to not have it hardcoded here.

File: tests/conftest.py [hunk 4]
```diff
@@ -67,117 +68,30 @@ async def client_auth(client_unauth: "AsyncClient") -> "AsyncClient":
...
+    db_sessionmaker = create_test_db_sessionmaker(engine)
+    await prepare_test_db(engine, db_sessionmaker, True)
+    return engine, db_sessionmaker
 
 
-@pytest.fixture()
-def db_sessionmaker(db_engine):
-    return sessionmaker(bind=db_engine, autoflush=False, future=True, expire_on_commit=False, class_=AsyncSession)
+@pytest.fixture(scope="session")
+def seed_db_sessionmaker(seed_db_engine: tuple["AsyncEngine", "sessionmaker"]) -> "sessionmaker":
```
Comment (gbdlin) [inline tests/conftest.py:80]: We don't have unseed DB anymore, we can drop `seed` from names everywhere.

**Comment by AAGaming00** [comment]
we should really merge this already

**Pull request merged**